src/AutoSavings/AutoSavings.sol

```diff
@@ -74,7 +70,9 @@ contract AutoSavings is ERC7579ExecutorBase, InitializableUniswapV3Integration {
         address account = msg.sender;
 
         // decode the data to get the tokens and their configurations
-        (ConfigWithToken[] memory _configs) = abi.decode(data, (ConfigWithToken[]));
+        (address swapRouter, uint24 swapRouterFee, ConfigWithToken[] memory _configs) =
+            abi.decode(data, (address, uint24, ConfigWithToken[]));
+        _initSwapRouter(swapRouter, swapRouterFee);
```

Comment:
shouldnt this fee also be variable?

Reply:
eg for stable pairs vs exotic pairs theres different liquidity on different fees right